### DISCUSSION TITLE
How to generate multiple non-html files using template-engine?

### DISCUSSION THREAD

**Discussion by cawa-93** [lumeland/lume#477] [Q&A]
I need to create several similar non-html files based on `docData`. I thought about using layouts for this, but [it is not possible](https://github.com/lumeland/lume/issues/474#issuecomment-1694493298). 

How can I manually call render the template engine I need to render content directly without layouts?
```js
// docs.tmpl.js
export default function * ({ docData }) {
	for (const doc of docData) {
		yield {
			url: `docs/${doc.title}.md`,
			content: renderTemplateEngineSomehow( 'docTemplate.njk' ), // 👈
		}
	}
}
```

**Comment by oscarotero** [comment]
The [`templateEngine` variable](https://lume.land/docs/core/multiple-template-engines/) allows to define a engine to render the page.

```js
// docs.tmpl.js
export default function * ({ docData }) {
	for (const doc of docData) {
		yield {
			url: `docs/${doc.title}.md`,
			content: doc.content,
                        templateEngine: "njk"
		}
	}
}
```

I'm thinking of allowing to render the layout for non-html pages only if the layout is defined directly in the page data (not inherited from a _data files).
